### PR TITLE
Toast for error handling module

### PR DESCRIPTION
**Pull request by Andu033**
I still need to style the button. I need a X icon. Also need to create the page in storybook.

### CODE REVIEW

File: src/toasts/toasts.stories.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { storiesOf } from "@storybook/html";
+import html from "./toasts.stories.html";
+import "./toasts.css";
+import ejs from "ejs";
+
+import { htmlStory } from "../story-helpers";
+import { showToast } from "./toasts.js";
+// storiesOf("OpenMRS Styleguide", module).add(
```
Comment (dkayiwa) [inline src/toasts/toasts.stories.js]: Did you intend to leave these in?

Comment (Andu033) [reply src/toasts/toasts.stories.js]: No.

Comment (Andu033) [reply src/toasts/toasts.stories.js]: I need to finish this file. I have to create a story for this toast. This comment is temporary.

File: src/toasts/toasts.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+export function showToast(description) {
```
Comment (joeldenning) [inline src/toasts/toasts.js]: Let's change `showToast` to take in an `options` object as an argument. That way we can make it possible for the user to configure things like timeout duration, coloration/styling of the toast, maybe "action button callback", etc.

File: src/toasts/toasts.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  innerDiv.setAttribute("id", "text");
+
+  innerDiv.appendChild(textnode);
+  toast.appendChild(innerDiv);
+  toast.append(divider);
+  toast.append(button);
+
+  var timeout = setTimeout(function() {
+    document.body.removeChild(toast);
+  }, 10000);
```
Comment (joeldenning) [inline src/toasts/toasts.js]: 10 seconds is too much. Usually somewhere around 4-6 is better.

File: src/toasts/toasts.stories.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+// );
+
+storiesOf("OpenMRS Styleguide", module).add("MyState", () => {
+  function init() {
+    var button = document.querySelector(".js-toggle");
+    button.addEventListener("click", showToast("This is a toast"));
+  }
+
+  document.addEventListener("DOMContentLoaded", init);
+  return '<button class="js-toggle">Toggle</button>';
```
Comment (joeldenning) [inline src/toasts/toasts.stories.js]: Please put the html into a separate file. See [here](https://github.com/openmrs/openmrs-esm-styleguide/blob/master/src/cards/cards.stories.js#L10) and [here](https://github.com/openmrs/openmrs-esm-styleguide/blob/master/src/cards/cards.stories.html) for example

**Comment by joeldenning** [comment]
Btw for the future you can create [draft pull requests](https://github.blog/2019-02-14-introducing-draft-pull-requests/) if you want to get feedback before something is finished. That helps the reviewer know that things might not be finalized.

File: src/toasts/toasts.css [hunk 5]
```diff
@@ -12,22 +12,34 @@
...
+
+#omrs-toast-btn {
+  display: block;
+  height: 100%;
+
+  float: right;
+}
+
 #omrs-toast #text {
   margin: 16px 12px 16px 12px;
```
Comment (fatmali) [inline src/toasts/toasts.css]: @Andu033 `rem` is better to use than `px`

File: src/toasts/toasts.css [hunk 6]
```diff
@@ -0,0 +1,45 @@
+#omrs-toast {
+  visibility: hidden;
+  max-width: 22rem;
+  height: auto;
+  margin: auto;
+  min-height: 3rem;
+  background-color: #323233;
+  color: var(--omrs-color-ink-white);
```
[the remark this answers is not in the document]
Comment (Andu033) [reply src/toasts/toasts.css]: yes. Here https://www.figma.com/file/h3WbqbfaZ1JE6Ww4G2mMyW/vs0.05-OpenMRS-Design-System?node-id=877%3A4627

File: src/toasts/toasts.stories.js (hunk 4 above)
Comment (Andu033) [reply src/toasts/toasts.stories.js]: I will modify that after I will figure out how to create a story for an react component or dom or what is this.

File: src/toasts/toasts.css (hunk 5 above)
Comment (Andu033) [reply src/toasts/toasts.css]: Ok. I will change

File: src/toasts/toasts.css [hunk 7]
```diff
@@ -0,0 +1,45 @@
+#omrs-toast {
```
[the remark this answers is not in the document]
Comment (Andu033) [reply src/toasts/toasts.css]: I modified the PR. You can review. I almost did al changes but the component doesn't work when I call the function in the console.

File: src/styleguide-entry.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import "./set-public-path.js";
```
[the remark this answers is not in the document]
Comment (Andu033) [reply src/styleguide-entry.js]: ok

File: src/toasts/toasts.css [hunk 9]
```diff
@@ -0,0 +1,40 @@
+.omrs-toast {
+  visibility: hidden;
```
Comment (joeldenning) [inline src/toasts/toasts.css]: Why is this hidden?

Comment (Andu033) [reply src/toasts/toasts.css]: It was from the previous implementation. I removed that and it works. Needs to be styled and to add options for button or without close button. A question. Greg designed a retry button. Is this in our scope or just display "Error" in a toast with a close button is all we want?

Comment (joeldenning) [reply src/toasts/toasts.css]: For now, don't worry about retries.

> It was from the previous implementation. I removed that and it works. 

The `visibility: hidden;` is not removed. Maybe you forgot to push that code?

File: src/toasts/toasts.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import ReactDOM from "react-dom";
+
+const toastsContainer = document.createElement("div");
+document.body.appendChild(toastsContainer);
+
+export function showToast(options) {
```
Comment (fatmali) [inline src/toasts/toasts.js]: @Andu033 @joeldenning is it possible to use jsdoc/types here in order to know the signature of the options being expected?

File: src/toasts/toasts.stories.html [hunk 11]
```diff
@@ -0,0 +1,21 @@
+<article class="stories-spacing">
+  <h2>OpenMRS Spacing, Padding, and Margin</h2>
+  The OpenMRS styleguide provides css classes for padding and margin.
+  <section>
+    <h3>Padding</h3>
+
+    <div class="classname-columns">
```
Comment (fatmali) [inline src/toasts/toasts.stories.html]: @Andu033 could you also add a tab on how to use the showToast in JSX ? check the latest commit on how to achieve this

File: src/toasts/toasts.js (hunk 10 above)
Comment (joeldenning) [reply src/toasts/toasts.js]: We could add typescript to esm-styleguide. I think it's okay to start without typescript and add it in as a future PR, if we want it.

File: src/toasts/toasts.stories.html (hunk 11 above)
Comment (Andu033) [reply src/toasts/toasts.stories.html]: I think i figured a way but the onclick doesn't work. But if i change to other story and come back the toast pop up.

File: src/toasts/toasts.css (hunk 9 above)
Comment (Andu033) [reply src/toasts/toasts.css]: I didn't push the changes.

File: src/toasts/toasts.stories.js (hunk 4 above)
Comment (Andu033) [reply src/toasts/toasts.stories.js]: Can you help me with stylling? Cant center that X icon in the button and can't resize it.

File: src/toasts/toasts.stories.js [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { storiesOf } from "@storybook/html";
+import html from "./toasts.stories.html";
+import "./toasts.css";
+import ejs from "ejs";
+
+import { htmlStory } from "../story-helpers";
+
+storiesOf("OpenMRS Styleguide", module).add("Toasts", () => {
+  return htmlStory(ejs.render(html, { text: "This is a toast" }));
```
Comment (joeldenning) [inline src/toasts/toasts.stories.js]: You're not using the `text` inside of the template. Should be removed.

```suggestion
  return htmlStory(ejs.render(html));
```

File: .storybook/custom-decorators/toast.decorator.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import { useEffect } from "@storybook/client-api";
+import { showToast } from "../../src/toasts/toasts.js";
+
+function getButton() {
+  return document.querySelectorAll(".toast-clicked");
+}
+
+export const withToast = storyFn => {
```
Comment (joeldenning) [inline .storybook/custom-decorators/toast.decorator.js]: Pretty clever solution. Nice work 👍

**Pull request merged**